src/rules/select-style.ts

```diff
@@ -0,0 +1,62 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils'
+
+import { docsUrl, pipeableSelect, storeSelect } from '../utils'
+
+export const ruleName = 'select-style'
+
+export const methodSelectMessageId = 'storeSelect'
+export const operatorSelectMessageId = 'operatorSelect'
+
+export type MessageIds =
+  | typeof methodSelectMessageId
+  | typeof operatorSelectMessageId
+
+export const OPERATOR = 'operator'
+export const METHOD = 'method'
+
+type Options = [string]
```

Comment:
WDYT about the following syntax for it? I've mostly seen it this way

```suggestion
type Options = [{
  mode: string
}]
```